services/addons/images/firmware_manager/upgrade_scheduler/upgrade_scheduler.py

```diff
@@ -81,18 +75,22 @@ def start_tasks_from_queue():
         try:
             rsu_upgrade_info = upgrade_queue_info[rsu_to_upgrade]
             del upgrade_queue_info[rsu_to_upgrade]
-            p = Popen(
-                [
-                    "python3",
-                    f'/home/{manufacturer_upgrade_scripts[rsu_upgrade_info["manufacturer"]]}',
-                    f"'{json.dumps(rsu_upgrade_info)}'",
-                ],
-                stdout=DEVNULL,
-            )
-            rsu_upgrade_info["process"] = p
+
+            # Begin the firmware upgrade using the Upgrade Runner API
+            upgrade_runner_endpoint = os.environ.get("UPGRADE_RUNNER_ENDPOINT", "UNDEFINED")

```

Comment:
If the upgrade runner endpoint is undefined should we throw an exception or handle that here?

Reply:
We could throw an exception.

Reply:
Addressed